prestoadmin/yarn_slider/server.py

```diff
@@ -248,3 +250,79 @@ def mk_data_dir():
              'user': app_config.get_user(),
              'group': app_config.get_group()})
     return sudo(command)
+
+
+def get_slider_status(conf):
+    status_cmd = '%s status %s' % (get_slider_bin(conf), conf[APP_INST_NAME])
+    with hide('running', 'stdout', 'stderr'):
+        status_json = degarbage_json(run_slider(status_cmd, conf))
+    return json.loads(status_json)
+
+
+ROLENAME_COORDINATOR = 'COORDINATOR'
+ROLENAME_WORKER = 'WORKER'
+
+
+def get_instances(slider_status, rolename):
+    return slider_status['instances'][rolename]
+
+
+def identity(x):
+    return x
+
+
+def get_coordinator(slider_status, filtr=identity):
+    coordinators = slider_status['status']['live'][ROLENAME_COORDINATOR]
```

Comment:
Unclear; this falls into the usual problem of getting something into a messed-up state long enough to determine the expected behavior. I'll dig into the Slider code and see if I can pull out something useful.